back/engines/commercial/content_builder/app/models/content_builder/layout.rb

```diff
@@ -19,6 +19,34 @@ module ContentBuilder
   class Layout < ApplicationRecord
     belongs_to :content_buildable, polymorphic: true
 
+    before_validation :sanitize_craftjs_jsonmultiloc
+
     validates :content_buildable, :code, presence: true
+    validates :craftjs_jsonmultiloc, multiloc: { presence: false, value_type: Hash }
+    validate :validate_craftjs_jsonmultiloc
+
+    private
+
+    def validate_craftjs_jsonmultiloc
+      validate_whitelisted_iframe_urls
+    end
+
+    def validate_whitelisted_iframe_urls
+      craftjs_jsonmultiloc.each do |locale, json|
+        json.select do |key, elt|
```

Comment:
I think that the 2 loops can be reduced to 1 loop when using `each` and using `next` if the condition on line 37 evaluates to `false`.